interfaces/builtin/fwupd.go

```diff
@@ -65,17 +65,26 @@ const fwupdPermanentSlotAppArmor = `
   # Allow fwupd to access system information
   /sys/devices/virtual/dmi/id/product_name r,
   /sys/devices/virtual/dmi/id/sys_vendor r,
+  /sys/firmware/dmi/tables/ r,
+  /sys/firmware/dmi/tables/** r,
 
   # Allow read/write access for efivarfs filesystem
   /sys/firmware/efi/efivars/ r,
   /sys/firmware/efi/efivars/** rw,
 
   # Allow write access for efi firmware updater
+  /boot/efi/{,**/} r,
+  /boot/efi/EFI/ubuntu/fwupd*.efi* rw,
```

Comment:
This is much better, thanks! FYI, I would not be opposed to this rule for future proofing: `/boot/efi/EFI/*/fwupd*.efi* rw,`, but it is fine as is if it gives you what you need.

Reply:
Now only two lines are left for purpose. We may change it again if there is something changing in future, e.g. fwupd expects to read ID from /etc/os-release.

Reply:
Actually current version of fwupd does read from `/etc/os-release` to figure out what to put in this path.